x-pack/auditbeat/module/system/process/quark_provider_linux.go

```diff
@@ -0,0 +1,330 @@
+	if process.Exit.Valid {
+		event.RootFields.Put("process.exit_code", process.Exit.ExitCode)
+	}
+	if !process.Proc.Valid {
+		return event, true
+	}
+
+	//
+	// Code below can rely on Proc
+	//
+
+	// Ids
+	event.RootFields.Put("process.parent.pid", process.Proc.Ppid)
+	startTime := time.Unix(0, int64(process.Proc.TimeBoot))
+	if ms.HostID() != "" {
+		// TODO unify with sessionview and guarantee loss of precision
+		event.RootFields.Put("process.entity_id",
+			entityID(ms.HostID(), int(process.Pid), startTime))
+	}
+	event.RootFields.Put("process.start", startTime)
+	event.RootFields.Put("user.id", process.Proc.Uid)
+	event.RootFields.Put("user.group.id", process.Proc.Gid)
+	event.RootFields.Put("user.effective.id", process.Proc.Euid)
+	event.RootFields.Put("user.effective.group.id", process.Proc.Egid)
+	event.RootFields.Put("user.saved.id", process.Proc.Suid)
+	event.RootFields.Put("user.saved.group.id", process.Proc.Sgid)
+	if us, err := user.LookupId(strconv.FormatUint(uint64(process.Proc.Uid), 10)); err == nil {
+		event.RootFields.Put("user.name", us.Username)
+		username = us.Username
+	}
+	if group, err := user.LookupGroupId(strconv.FormatUint(uint64(process.Proc.Gid), 10)); err == nil {
+		event.RootFields.Put("user.group.name", group.Name)
+	}
+	// Tty things
+	event.RootFields.Put("process.interactive",
+		tty.InteractiveFromTTY(tty.TTYDev{
+			Major: process.Proc.TtyMajor,
+			Minor: process.Proc.TtyMinor,
+		}))
+	if process.Proc.TtyMajor != 0 {
+		event.RootFields.Put("process.tty.char_device.major", process.Proc.TtyMajor)
+		event.RootFields.Put("process.tty.char_device.minor", process.Proc.TtyMinor)
+	}
+	// Capabilities
+	capEffective, _ := capabilities.FromUint64(process.Proc.CapEffective)
+	if len(capEffective) > 0 {
+		event.RootFields.Put("process.thread.capabilities.effective", capEffective)
+	}
+	capPermitted, _ := capabilities.FromUint64(process.Proc.CapPermitted)
+	if len(capPermitted) > 0 {
+		event.RootFields.Put("process.thread.capabilities.permitted", capPermitted)
+	}
+	// If we are in the same mount namespace of the process, hash
+	// the file. When quark is running on kprobes, there are
+	// limitations concerning the full path of the filename, in
+	// those cases, the path won't start with a slash.
+	if process.Proc.MntInonum == ms.selfMntNsIno && len(process.Filename) > 0 && process.Filename[0] == '/' {
+		hashes, err := ms.cachedHasher.HashFile(process.Filename)
+		if err != nil {
+			processErr = fmt.Errorf("failed to hash executable %v for PID %v: %w",
+				process.Filename, process.Pid, err)
+			ms.log.Error(processErr.Error())
+		} else {
+			for hashType, digest := range hashes {
+				fieldName := "process.hash." + string(hashType)
+				event.RootFields.Put(fieldName, digest)
+			}
+		}
+	} else {
+		ms.log.Debugf("skipping hash %s (inonum %d vs %d)\n", process.Filename, process.Proc.MntInonum, ms.selfMntNsIno)
+	}
+	event.RootFields.Put("quark", true) // XXX REMOVE ME
```

Comment:
```suggestion
```
Ready to remove this?

Reply:
I didn't forget, I swear! :rofl:. I'm keeping this until the last commit, basically it's what I'm using on kibana since I have vanilla auditbeat + auditbeat@quark populating the same table.

Reply:
oh that's neato, I'll start using that, many thanks.
Removed in ae5a34f17bb1d2c879fdb9b72a00c41e35d04ae3